### PR TITLE
New nft proofs + commit

### PR DESCRIPTION
**Pull request by mikiquantum**
Remaining here is:
- Naming changes PaymentObligation -> InvoiceUnpaid
- TokenURI removal from mint call
- Update deployed addresses
- Regenerate Identities?

### CODE REVIEW

File: nft/ethereum_payment_obligation.go [hunk 1]
```diff
@@ -2,11 +2,16 @@ package nft
...
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
 	"math/big"
 	"time"
 
+	"github.com/centrifuge/go-centrifuge/utils/byteutils"
+
+	"github.com/centrifuge/go-centrifuge/utils/stringutils"
+
```
Comment (vedhavyas) [inline nft/ethereum_payment_obligation.go]: import spacing

File: nft/ethereum_payment_obligation.go [hunk 2]
```diff
@@ -68,6 +73,21 @@ func newEthereumPaymentObligation(
...
 
+func (s *ethereumPaymentObligation) filterMintProofs(docProof *documents.DocumentProof) *documents.DocumentProof {
+	// Compact properties
+	var nonFilteredProofsLiteral = [][]byte{append(documents.CompactProperties(documents.DRTreePrefix), documents.CompactProperties(documents.SigningRootField)...)}
+	// Byte array Regex - (signatureTreePrefix + signatureProp) + Index[up to 104 characters (52bytes*2)] + Signature key
+	m0 := append(documents.CompactProperties(documents.SignaturesTreePrefix), []byte{0, 0, 0, 1}...)
+	var nonFilteredProofsMatch = []string{fmt.Sprintf("%s(.{104})%s", hex.EncodeToString(m0), hex.EncodeToString([]byte{0, 0, 0, 4}))}
+
+	for i, p := range docProof.FieldProofs {
+		if !byteutils.ContainsBytesInSlice(nonFilteredProofsLiteral, p.GetCompactName()) && !stringutils.ContainsBytesMatchInSlice(nonFilteredProofsMatch, p.GetCompactName()) {
```
Comment (vedhavyas) [inline nft/ethereum_payment_obligation.go]:
```suggestion
		if !byteutils.ContainsBytesInSlice(nonFilteredProofsLiteral, p.GetCompactName()) && !byteutils.ContainsBytesMatchInSlice(nonFilteredProofsMatch, p.GetCompactName()) {
```

File: utils/stringutils/utils.go [hunk 3]
```diff
@@ -19,3 +23,38 @@ func RemoveDuplicates(strs []string) []string {
...
+}
+
+// ContainsBytesMatch takes a hex string match regex and a []byte value and tries to find a match
+func ContainsBytesMatch(match string, val []byte) bool {
+	enc := hex.EncodeToString(val)
+	return ContainsStringMatch(match, enc)
+}
+
+// ContainsBytesMatchInSlice takes a hex string match regex and a []byte value and tries to find a match
+func ContainsBytesMatchInSlice(slice []string, val []byte) bool {
```
Comment (vedhavyas) [inline utils/stringutils/utils.go]: these should be in byteutils, no?

Comment (vimukthi-git) [reply utils/stringutils/utils.go]: I don't mind either way,  so let's keep it this way.

File: nft/ethereum_payment_obligation.go (hunk 2 above)
Comment (vimukthi-git) [reply nft/ethereum_payment_obligation.go]: as explained

**Pull request merged**